tasks/create_distribution.yml

```diff
@@ -0,0 +1,32 @@
+- block:
+  - name: Create distribution
+    uri:
+      url: "http://{{ pulp_api_host }}:{{ pulp_api_port }}/pulp/api/v3/distributions/ansible/ansible/"
+      method: POST
+      body:
+        base_path: "{{ main_item.base_path }}"
+        name: "{{ main_item.name }}"
+        repository: "http://{{ pulp_api_host }}:{{ pulp_api_port }}{{ repository_created_response.json.pulp_href }}"
+      body_format: json
+      user: "{{ pulp_admin_username }}"
+      password: "{{ pulp_default_admin_password }}"
+      force_basic_auth: yes
+      status_code: 202
+    register: distribution_created_response
+    run_once: true
+
+  - name: Wait until the distribution task is complete
+    uri:
+      url: "http://{{ pulp_api_host }}:{{ pulp_api_port }}{{ distribution_created_response.json.task }}"
+      user: "{{ pulp_admin_username }}"
+      password: "{{ pulp_default_admin_password }}"
+      force_basic_auth: yes
+    register: distribution_created_task_response
+    until: distribution_created_task_response.json.state != 'waiting'
+    delay: 2
+    retries: 60
+
+  - debug:
+      msg: "Task for creating distribution did not reach a successful state. {{ distribution_created_task_response }}"
+    when: distribution_created_task_response.json.state != 'completed'
+  when: repository_exists_response.json.count == 0
```

Comment:
I know it's not directly related to your PR but since you are dealing with it it might make sense to address it here.

This condition is not fully accurate, it is limiting the creation of the distribution on the (non) existence of the repository rather than the distribution itself.

Say you run it once, it creates the reposttory and then something happens which leads to the distribution not being created. Fine, I'll re-run it. Now this whole block is skipped because the repository actually exist but not the disttribution. One ends up in a broken state.

IMO, this block should be conditionned to the result of an `name: Check if distribution exists` that apply the same pattern as repository existance check here https://github.com/chouseknecht/ansible_galaxy_config/blob/master/tasks/create_repositories.yml#L1-L8